### PR TITLE
48 hour followup list

### PR DESCRIPTION
**Pull request by divs1210**
https://www.pivotaltracker.com/story/show/163902345

### CODE REVIEW

File: app/controllers/patients_controller.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+class PatientsController < AdminController
+  before_action :set_patient, only: [:edit, :update, :cancel]
+
+  def index
+    authorize Patient, :index?
+    @patients_per_facility = policy_scope(Patient)
+                               .not_contacted
+                               .select { |patient| patient.blood_pressures.present? }
```
[the remark this answers is not in the document]
Comment (divs1210) [reply app/controllers/patients_controller.rb]: Not sure how to do this.

File: app/views/patients/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<h1>Confirm Patient Followup</h1>
+
+<%= form_tag(patient_path, method: 'put') do %>
+  Contacted:
+  <%= check_box(:patient, :contacted_by_counsellor,
```
[the remark this answers is not in the document]
Comment (divs1210) [reply app/views/patients/edit.html.erb]: - initial state: [nil, false]
- next states: [nil, true], ['some_reason', false]

this validation doesn't apply here

File: app/controllers/patients_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (divs1210) [reply app/controllers/patients_controller.rb]: we have added pagination via kaminari  (for the overdue list)

File: db/migrate/20190214075550_add_followup_fields_to_patient.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class AddFollowupFieldsToPatient < ActiveRecord::Migration[5.1]
+  def change
```
[the remark this answers is not in the document]
Comment (divs1210) [reply db/migrate/20190214075550_add_followup_fields_to_patient.rb]: added pagination for now, can think about this later

File: app/views/patients/cancel.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<h1>Cancel Followup</h1>
+
+<%= form_tag(patient_path(@patient), method: 'put') do %>
+  Cancel Reason:
+  <%= select :patient, :could_not_contact_reason, Appointment.cancel_reasons.keys, selected: 'other' %>
```
[the remark this answers is not in the document]
Comment (divs1210) [reply app/views/patients/cancel.html.erb]: done

File: db/migrate/20190214075550_add_followup_fields_to_patient.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+    add_column :patients, :contacted_by_counsellor, :boolean, default: false
```
Comment (govindkrjoshi) [inline db/migrate/20190214075550_add_followup_fields_to_patient.rb]: This is better tracked by creating records of the communication model. 
However, only a User can create a Communication record. We'll either have to make communication user polymorphic, or preferably use the new combined User and Admin model after the roles and permissions refactoring.

File: app/models/patient.rb [hunk 6]
```diff
@@ -48,4 +60,25 @@ def current_age
...
     end
   end
+
+  def self.not_contacted
+    where(contacted_by_counsellor: false)
+      .where(could_not_contact_reason: nil)
+      .where('device_created_at <= ?', 2.days.ago)
+  end
+
+  def call_result=(new_call_result)
```
Comment (govindkrjoshi) [inline app/models/patient.rb]: This should be a seperate setter method. Expecting other attributes to change when assigning a value using `=` is very unintuitive.

File: app/models/patient.rb [hunk 7]
```diff
@@ -48,4 +60,25 @@ def current_age
       age + years_since_update
     end
   end
+
+  def self.not_contacted
```
Comment (govindkrjoshi) [inline app/models/patient.rb]: This can be a scope.

**Pull request merged**